tests/csharp-stellar-base.Tests/OperationTests.cs

```diff
@@ -9,66 +9,93 @@ namespace csharp_stellar_base.Tests
+            Assert.AreEqual(asset.Type, operation.Asset.Type);
+            Assert.AreEqual(limit, operation.Limit);
+            Assert.AreEqual("AAAAAQAAAAC7JAuE3XvquOnbsgv2SRztjuk4RoBVefQ0rlrFMMQvfAAAAAYAAAABRVVSAAAAAAC7JAuE3XvquOnbsgv2SRztjuk4RoBVefQ0rlrFMMQvfAAAAAAAAABk",
+                    operation.ToXdrBase64());
+
             Stellar.Generated.Operation xdr = operation.ToXdr();
             ChangeTrustOperation parsedOperation = Stellar.ChangeTrustOperation.FromXdr(xdr);
 
             Assert.AreEqual(source.Address, parsedOperation.SourceAccount.Address);
-            Assert.AreEqual("EUR", Encoding.ASCII.GetString(parsedOperation.Asset.AlphaNum4.AssetCode));
-            Assert.AreEqual(Stellar.Generated.AssetType.AssetTypeEnum.ASSET_TYPE_CREDIT_ALPHANUM4,
-                            parsedOperation.Asset.Discriminant.InnerValue);
+            Assert.AreEqual("EUR", parsedOperation.Asset.Code);
+            Assert.AreEqual(source.Address, parsedOperation.Asset.Issuer.Address);
             Assert.AreEqual(limit, parsedOperation.Limit);
-            Assert.AreEqual("AAAAAQAAAAC7JAuE3XvquOnbsgv2SRztjuk4RoBVefQ0rlrFMMQvfAAAAAYAAAABRVVSAAAAAAC7JAuE3XvquOnbsgv2SRztjuk4RoBVefQ0rlrFMMQvfAAAAAAAAABk",
+        }
+
+        [TestMethod]
+        public void PaymentOperation()
+        {
+            // GC5SIC4E3V56VOHJ3OZAX5SJDTWY52JYI2AFK6PUGSXFVRJQYQXXZBZF
+            KeyPair source = KeyPair.FromSeed("SC4CGETADVYTCR5HEAVZRB3DZQY5Y4J7RFNJTRA6ESMHIPEZUSTE2QDK");
+            // GDW6AUTBXTOC7FIKUO5BOO3OGLK4SF7ZPOBLMQHMZDI45J2Z6VXRB5NR
+            KeyPair destination = KeyPair.FromSeed("SDHZGHURAYXKU2KMVHPOXI6JG2Q4BSQUQCEOY72O3QQTCLR2T455PMII");
+
+            Asset asset = new Stellar.Asset();
+            long amount = 1000;
+
+            PaymentOperation operation = new PaymentOperation.Builder(destination, asset, amount)
+                .SetSourceAccount(source)
+                .Build();
+
+            Assert.AreEqual(source.Address, operation.SourceAccount.Address);
+            Assert.AreEqual(destination.Address, operation.Destination.Address);
+            Assert.AreEqual(asset.Type, operation.Asset.Type);
+            Assert.AreEqual(amount, operation.Amount);
+            Assert.AreEqual(
+                    "AAAAAQAAAAC7JAuE3XvquOnbsgv2SRztjuk4RoBVefQ0rlrFMMQvfAAAAAEAAAAA7eBSYbzcL5UKo7oXO24y1ckX+XuCtkDsyNHOp1n1bxAAAAAAAAAAAAAAA+g=",
                     operation.ToXdrBase64());
+
+            Stellar.Generated.Operation xdr = operation.ToXdr();
+            PaymentOperation parsedOperation = Stellar.PaymentOperation.FromXdr(xdr);
+
+            Assert.AreEqual(source.Address, parsedOperation.SourceAccount.Address);
+            Assert.AreEqual(destination.Address, parsedOperation.Destination.Address);
+            Assert.AreEqual(asset.Type, parsedOperation.Asset.Type);
+            Assert.AreEqual(amount, parsedOperation.Amount); 
         }
     }
```

Comment:
I'm missing the unhappy paths here, try to create an assert that triggers every error you wrote in the code.